scipy/optimize/tests/test_zeros.py

```diff
@@ -302,6 +315,69 @@ def test_root_scalar_fail(self):
         with pytest.raises(ValueError):
             root_scalar(f1, method='halley', fprime2=f1_2, x0=3, xtol=1e-6)  # no fprime
 
+    def test_array_chandrupatla(self):
+        """test _chandrupatla with array, similar to newton method"""
+
+        k = np.arange(1,8)
+        result = zeros._chandrupatla(self.f3, 0, 3, args=(k,))
+        x_expected = ([1., 1.41421356, 1.73205081,
+                      2., 2.23606798,2.44948974,
+                      2.64575131])
+        assert_allclose(result.x, x_expected)
+
+    def test_chandrupatla_newton_similar(self):
+        """test that _chandrupatla is giving a similar result to newton"""
+
+        k = np.arange(1,8)
+
+        result = zeros._chandrupatla(self.f3, 0, 3, args=(k[0],))
+        expected = zeros.newton(self.f3, x0=0, x1=3, args=(k[0],))
+        assert_allclose(result.x, expected)
+
+    def test_chandrupatla_brent_similar(self):
+        """test than chandrupatla gives equal scalar result to brentq"""
+        def f3(x):
+            return (x**3 - 1)
+
+        result = zeros._chandrupatla(f3,0,3,maxiter=50)
+        expected = zeros.brentq(f3,0,3)
+
+        assert_allclose(result.x, expected)
+
+    def test_chandrupatla_np_vectorize(self):
+        """test that _chandrupatla handles input with np.vectorize"""
+
+        k = np.arange(1,8)
+
+        res = zeros._chandrupatla(self.f3,0,3,
+                                  args=(k,),
+                                  maxiter=50)
+
+        vec_res = zeros._chandrupatla(np.vectorize(self.f3),0,3,
+                                  args=(k,),
+                                  maxiter=50)
+
+        assert_allclose(res.fun, vec_res.fun)
+        assert_allclose(res.x, vec_res.x)
+
+    def test_chandrupatla_broadcasting(self):
+        """test that _chandrupatla broadcasts inputs correctly"""
+
+        x0 = np.zeros((2,1,1))
+        x1 = np.full((1,2,1), 3)
+        k = np.array([[[1, 2]]])
+
+        res = zeros._chandrupatla(self.f3,x0=x0,x1=x1,
+                                  args=(k,),
+                                  maxiter=50)
+
+        expected = ([[[1., 1.41421356],
+                      [1., 1.41421356]],
+                     [[1., 1.41421356],
+                      [1., 1.41421356]]])
+
+        assert_allclose(res.x, expected)
```

Comment:
This is a good test, but it would be great to combine it with the vectorization test I suggest above. Does that make sense? The vectorization test shows that looping over multiple calls to `_chandrupatla` is the same as a single, vectorized call to `_chandrupatla`; we'd just augment that test by making the single, vectorized call one that involves broadcasting like this one. Please make it so that each output in `expected` is different (so that we would be able to tell if the outputs were out of order).